### DISCUSSION TITLE
Proper surname prefix support

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cms#9398] [Ideas]
> Created by: Bob Olde Hampsink (b.oldehampsink@itmundi.nl) on 2015/04/30 06:39:01 +0000
> Votes at time of UserVoice import: 4

-----------------------------------

In The Netherlands (and many other countries), we have surname prefixes. For example, you have Robin van Persie.

You enter it in Craft like this:
First Name: Robin
Last Name: van Persie

But when you want to sort, you (or atleast dutchies) expect to find the last name in the P section, not the V.

Could a surname prefix be supported in Craft?

**Comment by angrybrad** [comment]
> Posted by Christian (seelbach@gmx.de) on 2015/03/09 00:49:13 +0000
-----------------------------------

Do you want an extra field to solve this? Or should this happen automatically, so you&#39;d define a list of words being ignored when listing users or entries (remove articles etc. !) alphabetically?

**Comment by brandonkelly** [comment]
Researching this a bit. According to https://en.wikipedia.org/wiki/Tussenvoegsel, tussenvoegsels are _ignored_ when sorting by last name in the Netherlands (and [same with _von_ in Germany](https://en.wikipedia.org/wiki/Von), but they are _included_ when sorting by last name in Belgium.

I think that rules out giving users a dedicated “Preposition” or “Surname Prefix” field, as that would encourage _all_ users to fill it in if they have one, but that would result in surnames being sorted _incorrectly_ if you’re in Belgium.

Instead we should add a “Include surname prefixes when sorting users by last name” setting, which determines whether common surname prefixes should be included when sorting by last name.

So now I need to know where that should live. Which of these makes the most sense?

1. A global setting (either general config or in **Settings** → **Users**)
2. A user preference, enabled by default
3. Both (global setting determines the default user setting value)

I think I’m probably leaning toward option 3, but would appreciate some feedback.

cc @boboldehampsink @SayChi

**Comment by leevigraham** [comment]
> So now I need to know where that should live. Which of these makes the most sense?

If you're going to go down this path there's probably a few more things to consider:

* Honorific Prefix (Name Prefix)
* Honorific Suffix (Name Suffix)
* Additional Name

For reference:

* https://schema.org/Person
* https://www.w3.org/TR/WCAG21/#input-purposes (worth adding these to the front end form examples in the docs - @mattstein )
* VCARD Spec: https://datatracker.ietf.org/doc/html/rfc2426#section-3.1.2

In some cases it's also preferable just to show a single `Name` field eg: Display Name (Twitter).

I think Formie has a nice UI for this (although it's missing Honorific Suffix // @engram-design ):


https://user-images.githubusercontent.com/25124/121274220-64554600-c90d-11eb-949d-a1f47e899f4f.mov

**Comment by boboldehampsink** [comment]
Woah old issue :-) your suggestions look good @brandonkelly

**Comment by SayChi** [comment]
So if I understand it well, this would mean in the Netherlands we keep inputting names the same as described in the issue, and we get that option to determine how to sort it (Dutch vs Belgian sorting)? That sounds pretty good to me, especially option 3 with the global setting with a user based overwrite.

That, combined with that event for overwriting .name (143c3400eefc5fcf9c39bf776502748ae91299c4), would make me a very happy Dutch guy :)

(the event is especially useful to fix up some older projects, as we regularly forgot to do [user.firstname, user.tussenvoegsel, user.lastname]|join(' ') instead of .name, so we could overwrite it site wide on old projects)

**Comment by michaelrog** [comment]
fwiw — The [Wordsmith plugin](https://plugins.craftcms.com/wordsmith) exposes some decent functions for parsing names:

https://wordsmith.docs.topshelfcraft.com/guide/names.html

On projects where there's a directory/listing of people, I usually use a `title` field to contain the full name, and then set up a Preparse field called `alphaName` to automatically store the surname (without its compound/prefix). (Or, working with users, I'll do the same, but parsing the alpha name from `getFullName()`, since users don't have titles.)
